iPetCare/Application/Services/ExaminationParameterValueService.cs

```diff
@@ -0,0 +1,192 @@
+
+            var dto = new GetAllExaminationParametersValuesDtoResponse()
+            {
+                ExaminationParametersValues = Mapper.Map<List<ExaminationParameterValueForGetAllExaminationParametersValuesDtoResponse>>(examinationParametersValues)
+            };
+
+            return new ServiceResponse<GetAllExaminationParametersValuesDtoResponse>(HttpStatusCode.OK, dto);
+        }
+
+        public async Task<ServiceResponse<GetExaminationParameterValueDtoResponse>> GetExaminationParameterValueAsync(Guid examinationParameterValueId)
+        {
+            if (CurrentlyLoggedUser == null)
+                return new ServiceResponse<GetExaminationParameterValueDtoResponse>(HttpStatusCode.Unauthorized);
+
+            var examinationParameterValue = await Context.ExaminationParameterValues.FindAsync(examinationParameterValueId);
+            if (examinationParameterValue == null)
+                return new ServiceResponse<GetExaminationParameterValueDtoResponse>(HttpStatusCode.NotFound);
+
+            var examination = await Context.Examinations.FindAsync(examinationParameterValue.ExaminationId);
+            if (examination == null)
+                return new ServiceResponse<GetExaminationParameterValueDtoResponse>(HttpStatusCode.BadRequest, "Nie istnieje takie badanie w bazie danych");
+
+            var pet = Context.Pets.Find(examination.PetId);
+
+            if (pet == null)
+                return new ServiceResponse<GetExaminationParameterValueDtoResponse>(HttpStatusCode.BadRequest, "Nie znaleziono zwierzaka");
+
+            if (!CanEditExaminationParameterValue(pet))
+                return new ServiceResponse<GetExaminationParameterValueDtoResponse>(HttpStatusCode.Forbidden);
+
+            var examinationParameter = await Context.ExaminationParameters.FindAsync(examinationParameterValue.ExaminationParameterId);
+            if (examinationParameter == null)
+                return new ServiceResponse<GetExaminationParameterValueDtoResponse>(HttpStatusCode.NotFound);
+
+            examinationParameterValue.Examination = examination;
+            examinationParameterValue.ExaminationParameter = examinationParameter;
+
+            var dto = Mapper.Map<GetExaminationParameterValueDtoResponse>(examinationParameterValue);
+
+            return new ServiceResponse<GetExaminationParameterValueDtoResponse>(HttpStatusCode.OK, dto);
+        }
+
+        public async Task<ServiceResponse<UpdateExaminationParameterValueDtoResponse>> UpdateExaminationParameterValueAsync(Guid examinationParameterValueId, UpdateExaminationParameterValueDtoRequest dto)
+        {
+            if (CurrentlyLoggedUser == null)
+                return new ServiceResponse<UpdateExaminationParameterValueDtoResponse>(HttpStatusCode.Unauthorized);
+
+            var examinationParameterValue = await Context.ExaminationParameterValues.FindAsync(examinationParameterValueId);
+            if (examinationParameterValue == null)
```

Comment:
Taka w sumie ciekawa sprawa, jeśli mamy podany parametr Guid, to mamy pewność, że on nie przyjmie nulla nigdy w życiu, bo wywali exception. Guid jest typem nienullowalnym :D 

Zamiast tego niepodany może przyjąc specjalną wartość Guid.Empty jeśli mnie pamięć nie myli. To właśnie coś takiego byśmy obsługiwali.

Reply:
Czyli trzeba coś tu zmieniać, bo nie rozumiem?
Tu przecież sprawdzamy czy istnieje examinationParameterValue o podanym Guid w bazie

Reply:
W teorii to nie musimy. Jakbyśmy chcieli żeby nasz kod był świetny to możemy wcześniej sprawdzić czy == Guid.Empty i wtedy zwrócić nie strzelając do bazy (Jeden request mniej [ZOBACZ JAK ODCIĄŻYŁ APLIKACJE JEDNYM PROSTYM SPOSOBEM] XD)